src/octopath.cpp

```diff
@@ -106,199 +153,333 @@ bool OctoClass::OctoRRG(const Eigen::Vector3d &p0,
-        ROS_INFO("[mapper] Trying A* from node %d to node %d!", 0,
-                             static_cast<int>(final_index));
-        obj_rrg.rrgraph_.Astar2(0, final_index, index_path);
-
-        // Populate final path
-        if (index_path.size() == 0) {
-            ROS_WARN("[mapper] RRG Error: Couldn't find solution through Astar!");
-            *n_rrg_nodes = n_nodes;
-            *plan_time = (ros::Time::now() - t0).toSec();
-            return false;
-        } else {
-            for (uint i = 0; i < index_path.size(); i++) {
-                sol_path.push_back(obj_rrg.rrgraph_.nodes_[index_path[i]].pos_);
-            }
-            // ROS_INFO("Path size: %d", static_cast<int>(sol_path.size()));
+    // Check whether we reached the goal
+    octomap::point3d current_pos = tree_inflated_.keyToCoord(current_key);
+    if (current_key == final_key) {
+      Eigen::Vector3d pos = Eigen::Vector3d(current_pos.x(), current_pos.y(), current_pos.z());
+      path->insert(path->begin(), pos);
+      while (current_key != initial_key) {
+        if (!indexed_free_keys.Key2Index(current_key, &current_index)) {
+          ROS_INFO("[mapper] Astar failed when retrieving path from initial to final point!");
+          return false;
         }
-    } else {
-        ROS_WARN("[mapper] RRG Error: Could not find a path from p0 to pf!");
-        *n_rrg_nodes = n_nodes;
-        *plan_time = (ros::Time::now() - t0).toSec();
-        return false;
+        // ROS_INFO("Come from: %d", int(current_index));
+        current_key = come_from[current_index];
+        current_pos = tree_inflated_.keyToCoord(current_key);
+        pos = Eigen::Vector3d(current_pos.x(), current_pos.y(), current_pos.z());
+        path->insert(path->begin(), pos);
+      }
+      break;
     }
 
-    // Prune results if requested
-    if (prune_result) {
-        this->PathPruning(sol_path, free_space_only, path);
-    } else {
-        *path = sol_path;
+    // Check neighbors
+    std::vector<octomap::OcTreeKey> neighbor_keys;
+    this->GetNodeNeighbors(current_key, node_sizes[current_index], &neighbor_keys);
+    const uint n_neighbors = neighbor_keys.size();
+    // ROS_INFO("Current index: %d (%f, %f, %f) has %d neighbors", int(current_index),
+    //           current_pos.x(), current_pos.y(), current_pos.z(), int(n_neighbors));
+
+    for (uint i = 0; i < n_neighbors; i++) {
+      // Get neighbor index
+      uint neighbor_index;
+      if (!indexed_free_keys.Key2Index(neighbor_keys[i], &neighbor_index)) {
+        ROS_INFO("[mapper] Astar failed: Error retrieving index for neighbor node!");
+        return false;
+      }
+
+      // Compute an added cost based on the distance between the neighbor and the nearest obstacle
+      const octomap::point3d neighbor_pos = tree_inflated_.keyToCoord(neighbor_keys[i]);
+      if (node_distance_to_wall[neighbor_index] < 0.0) {
+        node_distance_to_wall[neighbor_index] = this->NearestObstaclePathCost(neighbor_pos);
```

Comment:
I would rename `node_distance_to_wall` to `obstacle_path_cost` because just calling it the distance is misleading

Reply:
good catch! I repurposed the variable `node_distance_to_wall` without renaming it. I'll rename as you suggested